dask_expr/expr.py

```diff
@@ -623,6 +623,36 @@ def visualize(self, filename="dask-expr.svg", format=None, **kwargs):
         graphviz_to_file(g, filename, format)
         return g
 
+    def find_subtype(self, subtype: type) -> Set[Expr]:
```

Comment:
I'll suggest that this be an iterator, like `os.path.walk`.  I think that the path dependence might be interesting in the future.

Names?  🤔 I don't like `subtype`.  We're looking for instances of a type.  `find_operations`?  `find_??`

Reply:
>I'll suggest that this be an iterator, like os.path.walk. I think that the path dependence might be interesting in the future.

Could you explain this a bit more? Are you just saying that we should return an iterator? (I get the feeling you're saying more than that)

Reply:
I suggest that rather than return a set we yield results as we find them.

I suspect that what you have here is a depth first search ordering.  That's a sensible ordering, probably more sensible than `set`'s hashing ordering or whatnot.  

Yielding results just makes more intuitive sense to me for traversing operations like this

Reply:
It's kind of an aesthetic thing though I guess